sdk/storage/storage-blob/src/utils/BlobQuickQueryStream.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Readable } from "stream";
+
+import { AbortError, AbortSignal, AbortSignalLike } from "@azure/abort-controller";
+import { TransferProgressEvent } from "@azure/core-http";
+
+import { AvroReadableFromStream, AvroReader } from "../../../storage-internal-avro/src";
+import { BlobQueryError } from "../Clients";
+
+export interface BlobQuickQueryStreamOptions {
+  /**
+   * An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
+   *
+   * @type {AbortSignalLike}
+   * @memberof BlobQuickQueryStreamOptions
+   */
+  abortSignal?: AbortSignalLike;
+
+  /**
+   * Read progress event handler
+   *
+   * @memberof BlobQuickQueryStreamOptions
+   */
+  onProgress?: (progress: TransferProgressEvent) => void;
+
+  /**
+   * Callback to receive error events during the query operaiton.
+   *
+   * @memberof BlockBlobQueryOptions
+   */
+  onError?: (error: BlobQueryError) => void;
+}
+
+const ABORT_ERROR = new AbortError("The operation was aborted.");
+
+/**
+ * ONLY AVAILABLE IN NODE.JS RUNTIME.
+ *
+ * A Node.js BlobQuickQueryStream will internally parser avor data stream for blob query.
```

Comment:
>parser avor [](start = 50, length = 11)

Would better create work items for TODOs.

Reply:
Will do.